precompile/allowlist/event.go

```diff
@@ -0,0 +1,53 @@
+// Code generated
+// This file is a generated precompile contract config with stubbed abstract functions.
+// The file is generated by a template. Please inspect every code and comment in this file before use.
+
+package allowlist
+
+import (
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	AllowListEventGasCost = contract.LogGas + contract.LogTopicGas*2
+)
+
+// PackAdminAddedEvent packs the event into the appropriate arguments for AdminAdded.
+// It returns topic hashes and the encoded non-indexed data.
+func PackAdminAddedEvent(sender common.Address, account common.Address) ([]common.Hash, []byte, error) {
+	return AllowListABI.PackEvent("AdminAdded", sender, account)
+}
+
+// PackEnabledAddedEvent packs the event into the appropriate arguments for EnabledAdded.
+// It returns topic hashes and the encoded non-indexed data.
+func PackEnabledAddedEvent(sender common.Address, account common.Address) ([]common.Hash, []byte, error) {
+	return AllowListABI.PackEvent("EnabledAdded", sender, account)
+}
+
+// PackManagerAddedEvent packs the event into the appropriate arguments for ManagerAdded.
+// It returns topic hashes and the encoded non-indexed data.
+func PackManagerAddedEvent(sender common.Address, account common.Address) ([]common.Hash, []byte, error) {
+	return AllowListABI.PackEvent("ManagerAdded", sender, account)
+}
+
+// PackRoleRemovedEvent packs the event into the appropriate arguments for RoleRemoved.
+// It returns topic hashes and the encoded non-indexed data.
+func PackRoleRemovedEvent(sender common.Address, account common.Address) ([]common.Hash, []byte, error) {
+	return AllowListABI.PackEvent("RoleRemoved", sender, account)
+}
+
+func PackAllowListEvent(role Role, sender common.Address, account common.Address) ([]common.Hash, []byte, error) {
+	switch role {
+	case AdminRole:
+		return PackAdminAddedEvent(sender, account)
+	case ManagerRole:
+		return PackManagerAddedEvent(sender, account)
+	case EnabledRole:
+		return PackEnabledAddedEvent(sender, account)
+	case NoRole:
+		return PackRoleRemovedEvent(sender, account)
+	default:
+		panic("invalid role")
+	}
```

Comment:
can we return an error instead of panic-ing here? I'd prefer defense in depth from something that should never happen to a panic that will shut down the whole chain.